core/src/test/java/kafka/admin/ConfigCommandIntegrationTest.java

```diff
@@ -172,72 +138,210 @@ public void testDynamicBrokerConfigUpdateUsingZooKeeper() throws Exception {
+    @ClusterTests({
+            @ClusterTest(clusterType = Type.KRAFT),
+            @ClusterTest(clusterType = Type.CO_KRAFT)
+    })
+    public void testUpdateClusterWideConfigInKRaftThenShouldSuccessful() {
+        alterOpts = generateKraftAlterOpts();
+        Admin client = cluster.createAdminClient();
+
+        alterAndVerifyConfig(client, Optional.of(defaultBrokerId), singletonMap("log.flush.interval.messages", "100"));
+        alterAndVerifyConfig(client, Optional.of(defaultBrokerId), singletonMap("log.retention.bytes", "20"));
+        alterAndVerifyConfig(client, Optional.of(defaultBrokerId), singletonMap("log.retention.ms", "2"));
+
+        client.close();
+    }
+
+    @ClusterTests({
+            @ClusterTest(clusterType = Type.KRAFT),
+            @ClusterTest(clusterType = Type.CO_KRAFT)
+    })
+    public void testUpdatePerBrokerConfigInKRaftThenShouldSuccessful() {
+        alterOpts = generateKraftAlterOpts();
+        Admin client = cluster.createAdminClient();
+
+        alterAndVerifyConfig(client, Optional.empty(), singletonMap("ssl.truststore.type", "PKCS12"));
+        alterAndVerifyConfig(client, Optional.empty(), singletonMap("ssl.truststore.location", "/temp/test.jks"));
+        alterAndVerifyConfig(client, Optional.empty(), singletonMap("ssl.truststore.password", "password"));
+
+        client.close();
+    }
+
+    private void assertNonZeroStatusExit(Stream<String> args, Consumer<String> checkErrOut) {
+        AtomicReference<Integer> exitStatus = new AtomicReference<>();
+        Exit.setExitProcedure((status, __) -> {
+            exitStatus.set(status);
+            throw new RuntimeException();
+        });
+
+        String errOut = captureStandardMsg(new BrokerCommand(args));
+
+        checkErrOut.accept(errOut);
+        assertNotNull(exitStatus.get());
+        assertEquals(1, exitStatus.get());
+    }
+
+    private Stream<String> quorumArgs() {
+        return cluster.isKRaftTest()
+                ? Stream.of("--bootstrap-server", cluster.bootstrapServers())
+                : Stream.of("--zookeeper", ((ZkClusterInvocationContext.ZkClusterInstance) cluster).getUnderlying().zkConnect());
+    }
+
+    private void verifyConfig(KafkaZkClient zkClient, Optional<String> brokerId, Map<String, String>... configs) {
+        Properties entityConfigs = zkClient.getEntityConfigs("brokers", brokerId.orElse(ZooKeeperInternals.DEFAULT_STRING));
+        for (Map<String, String> config : configs) {
+            assertEquals(config, entityConfigs);
+        }
+    }
+
+    private void alterConfigWithZk(KafkaZkClient zkClient, Optional<String> brokerId, Map<String, String>... configs) {
+        String configStr = transferConfigMapsToString(configs);
+        ConfigCommand.ConfigCommandOptions addOpts =
+                new ConfigCommand.ConfigCommandOptions(toArray(alterOpts, entityOp(brokerId), asList("--add-config", configStr)));
+        ConfigCommand.alterConfigWithZk(zkClient, addOpts, adminZkClient);
```

Comment:
Could we convert `adminZkClient` to a local variable here? A class-level `adminZkClient` is error-prone in virtue of NPE